crates/fuel-core/src/state/historical_rocksdb.rs

```diff
@@ -297,20 +419,171 @@ where
         )
         .commit()?;
 
-        self.db
-            .commit_changes(&storage_transaction.into_changes())?;
+        storage_transaction.commit()?;
+
+        match self
+            .db
+            .commit_changes(&migration_transaction.into_changes())
+        {
+            Ok(()) => Ok(()),
+            Err(err) => {
+                tracing::error!(
+                    "Could not rollback historical rocksDB to height {}: {err:?}",
+                    err
+                );
+                // If the migration is not in progress, this function will effectively be a no-op
+                self.add_migration_changes(
+                    maybe_cumulative_changes_guard,
+                    cumulative_changes,
+                )?;
+                Err(err)
+            }
+        }
+    }
+
+    /// Migrates a ModificationHistory key-value pair from V1 to V2.
+    /// The migration is lazy, in that the changes necessary to perform
+    /// the migration from ModificationHistoryV1 to ModificationHistoryV2
+    /// are simply recorded, and will be flushed to the database when it
+    /// commits or rollbacks to a new height.
+    #[cfg(test)]
+    fn migrate_modifications_history_at_height(&self, height: u64) -> StorageResult<()> {
```

Comment:
Side question: Why do we want to implement a function in this PR? It can be part of the migration PR.


I think the migration process should work with the iterator of the historical modification table instead of fetching each modification height by height. In this case, you can easily fetch butch of modifications up to some limit, migrate them, and store them in one `Changes` type inside of the `migration_changes` by just locking it once. Then the migration process can wait until we take changes from `migration_changes` during `commit_changes`, and continue after.

With this solution, you only lock the `migration_changes` changes once for very fast operation. Plus, the iteration over the storage will be much faster(and it will work even if we do rollback or alter the history somehow because while you are holding an iterator over RocksDB, it will not lose the data).

Reply:
Sounds good. I can make the change, although probably it makes sense to do it in the migration PR. 
Btw in this PR that function is only for testing purposes, so if that's okay I would keep it and change the migration process using the rocksDB iterator in the follow-up.

Reply:
It seems this function is only used inside of the `migrate_modifications_history_works` tests that is dedicated to this function=D I think it is better to test in the PR where we really plan to have final migration